lib/client/offline-manager/index.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+var queue = require('./queue');
+var _ = require("lodash");
+var QUEUE_NAME = "wfm-file-uploads";
+
+var FileQueueManager = function($fh, fileClient) {
+  this.$fh = $fh;
+  this.uploadQueue = new queue(QUEUE_NAME);
+  this.fileClient = fileClient;
+  // Start processing uploads on startup
+  this.startProcessingUploads();
+  // Listen when client becomes online for uploads
+  var self = this;
+  document.addEventListener("online", function() {
+    self.startProcessingUploads();
+  });
+};
+
+FileQueueManager.prototype.removeFromQueue = function(object) {
+  return this.uploadQueue.removeItem(object);
+};
+
+FileQueueManager.prototype.addQueueItem = function(object) {
+  return this.uploadQueue.addItem(object);
+};
+
+FileQueueManager.prototype.startProcessingUploads = function() {
+  var queueItems = this.uploadQueue.restoreData().getItemList();
+  var promise;
+  var self = this;
+  if (queueItems && queueItems.length > 0) {
+    console.log("Processing offline file queue. Number of items to save: ", queueItems.length);
+    promise = this.saveFile(queueItems[0]).then(function() {
```

Comment:
here you should call this.saveFile(_.noop) otherwise you will upload first file`queueItems[0]` twice